vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -231,17 +264,31 @@ def _validate_column_type(self, targeted_data: pd.DataFrame) -> Literal["numeric
             )
 
     @staticmethod
-    def _get_min_max(targeted_data: pd.DataFrame) -> tuple[float, float]:
+    def _get_min_max(targeted_data: pd.DataFrame, current_value=None) -> tuple[float, float]:
         # Use item() to convert to convert scalar from numpy to Python type. This isn't needed during pre_build because
         # pydantic will coerce the type, but it is necessary in __call__ where we don't update model field values
         # and instead just pass straight to the Dash component.
-        return targeted_data.min(axis=None).item(), targeted_data.max(axis=None).item()
+        _min = targeted_data.min(axis=None).item()
+        _max = targeted_data.max(axis=None).item()
+
+        if current_value is not None:
+            current_value = current_value if isinstance(current_value, list) else [current_value]
+            _min = min(_min, *current_value)
+            _max = max(_max, *current_value)
+
+        return _min, _max
 
     @staticmethod
-    def _get_options(targeted_data: pd.DataFrame) -> list[Any]:
+    def _get_options(targeted_data: pd.DataFrame, current_value=None) -> list[Any]:
         # Use tolist() to convert to convert scalar from numpy to Python type. This isn't needed during pre_build
         # because pydantic will coerce the type, but it is necessary in __call__ where we don't update model field
         # values and instead just pass straight to the Dash component.
         # The dropna() isn't strictly required here but will be in future pandas versions when the behavior of stack
         # changes. See https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#whatsnew-210-enhancements-new-stack.
-        return np.unique(targeted_data.stack().dropna()).tolist()  # noqa: PD013
+
+        # Remove ALL_OPTION from the string or list of currently selected value for the categorical filters.
+        current_value = [] if current_value in (None, ALL_OPTION) else current_value
+        if isinstance(current_value, list) and ALL_OPTION in current_value:
+            current_value.remove(ALL_OPTION)
+
+        return np.unique(pd.concat([targeted_data.stack().dropna(), pd.Series(current_value)])).tolist()  # noqa: PD013
```

Comment:
@antonymilne do you have a preference or a strong opinion on why `_get_options` and `_get_min_max` are `@staticmethod`, but `_filter_between` and `_filter_isin` are pure functions?